store/src/java-test/com/zimbra/cs/service/admin/ContactBackupApiTest.java

```diff
@@ -0,0 +1,128 @@
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.common.soap.ContactBackupServer;
+import com.zimbra.common.soap.ContactBackupServer.Status;
+import com.zimbra.common.soap.Element;
+import com.zimbra.cs.account.Account;
+import com.zimbra.cs.account.Provisioning;
+import com.zimbra.cs.mailbox.MailboxManager;
+import com.zimbra.cs.mailbox.MailboxTestUtil;
+import com.zimbra.cs.service.mail.ServiceTestUtil;
+import com.zimbra.soap.JaxbUtil;
+import com.zimbra.soap.ZimbraSoapContext;
+import com.zimbra.soap.admin.message.ContactBackupRequest;
+import com.zimbra.soap.admin.message.ContactBackupRequest.Operation;
+import com.zimbra.soap.admin.message.ContactBackupResponse;
+import com.zimbra.soap.admin.type.ServerSelector;
+import com.zimbra.soap.admin.type.ServerSelector.ServerBy;
+
+import junit.framework.Assert;
+
+public class ContactBackupApiTest {
+    private static Provisioning prov = null;
+    private static final String DOMAIN_NAME = "zimbra.com";
+    private static final String BUG_NUMBER = "zcs3594";
+    private static final String ADMIN = "admin_" + BUG_NUMBER + "@" + DOMAIN_NAME;
+    private static Account admin = null;
+
+    @BeforeClass
+    public static void init() throws Exception {
+        MailboxTestUtil.initServer();
+        prov = Provisioning.getInstance();
+
+        Map<String, Object> attrs = Maps.newHashMap();
+        attrs.put(Provisioning.A_zimbraId, UUID.randomUUID().toString());
+        attrs.put(Provisioning.A_zimbraIsAdminAccount, true); // set admin account
+        prov.createAccount(ADMIN, "secret", attrs);
+        admin = prov.getAccountByName(ADMIN);
+        MailboxManager.getInstance().getMailboxByAccount(admin);
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        MailboxTestUtil.clearData();
+    }
+
+    @Test
+    public void testContactBackupApiWithStart() throws Exception {
+        ContactBackupRequest cbReq = new ContactBackupRequest();
+        cbReq.setOp(Operation.start);
+        cbReq.addServer(new ServerSelector(ServerBy.name, "test1.com"));
+        cbReq.addServer(new ServerSelector(ServerBy.name, "test2.com"));
+        cbReq.addServer(new ServerSelector(ServerBy.name, "test3.com"));
+        Element request = JaxbUtil.jaxbToElement(cbReq);
+        Element response  = null;
+
+        try {
+            response = new MockContactBackup().handle(request, ServiceTestUtil.getRequestContext(admin));
+        } catch (ServiceException se) {
+            Assert.fail("ServiceException must not be thrown.");
+        }
+        if (response == null) {
+            Assert.fail("Response must be received.");
+        }
+        ContactBackupResponse cbResp = JaxbUtil.elementToJaxb(response);
+        Assert.assertNotNull(cbResp.getServers());
+        List<ContactBackupServer> servers = cbResp.getServers();
+        for (ContactBackupServer server : servers) {
+            Assert.assertEquals(server.getStatus(), Status.started);
```

Comment:
Can we test something which actual ContactBackup executes and is feasible to unit test? Something like what it returns if op is not passed etc.

Reply:
Nope. we can not do actual testing in unit tests, because it needs running server instance, which is not possible in case of unit tests.